qiime2/sdk/usage.py

```diff
@@ -922,6 +922,146 @@ def init_format(self, name: str,
         """
         return self._usage_variable(name, factory, 'format')
 
+    def _replace_url_epoch(self, url):
+        # Obtaining epoch modeled on qiime2.metadata.io.MetadataFileError
+        import qiime2
+
+        epoch = qiime2.__release__
+        return url.replace('{epoch}', epoch)
+
+    def _request_url(self, url):
+        import requests
+
+        try:
+            data = requests.get(url)
+            data.raise_for_status()
+        except requests.exceptions.RequestException as ex:
+            raise ValueError(
+                'Could not obtain URL: %s\n Requests exception: %s' %
+                (url, str(ex)))
+
+        return data
+
+    def init_artifact_from_url(self, name: str, url: str,
+                               replace_url_epoch: bool = True
```

Comment:
I like this too - on the off-chance that we ever don't need the epoch in a given artifact URL, this will be helpful.